llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -1567,6 +1573,36 @@ bool RISCVInstrInfo::isAsCheapAsAMove(const MachineInstr &MI) const {
   return MI.isAsCheapAsAMove();
 }
 
+bool RISCVInstrInfo::isReallyTriviallyReMaterializable(
```

Comment:
This is a correctness predicate and you've added performance heuristics; this should go somewhere else